io-engine/src/subsys/nvmf/subsystem.rs

```diff
@@ -233,57 +235,36 @@ impl NvmfSubsystem {
 
     /// Subsystem event handlers.
     extern "C" fn nvmf_subsystem_event_handler(
-        subsystem: *mut spdk_nvmf_subsystem,
+        subsys: *mut spdk_nvmf_subsystem,
         event: spdk_nvmf_subsystem_event,
         ctx: *mut c_void,
         _cb_arg: *mut c_void,
     ) {
-        let subsystem = NvmfSubsystem::from(subsystem);
-        let nqn = subsystem.get_nqn();
-        let nexus = nexus_lookup_nqn(&nqn);
+        let s = NvmfSubsystem::from(subsys);
         let event = NvmfSubsystemEvent::from_cb_args(event, ctx);
 
-        debug!("NVMF subsystem event {subsystem:?}: {event:?}");
-
-        match event {
-            NvmfSubsystemEvent::HostConnect(ctrlr) => {
-                info!(
-                    "Subsystem '{nqn}': host connected: '{host}'",
-                    host = ctrlr.hostnqn()
-                );
-
-                if let Some(nex) = nexus {
-                    nex.add_initiator(&ctrlr.hostnqn());
-                    subsystem.host_connect_nexus(ctrlr);
-                } else {
-                    subsystem.host_connect_replica(ctrlr);
-                }
-            }
-            NvmfSubsystemEvent::HostDisconnect(ctrlr) => {
-                info!(
-                    "Subsystem '{nqn}': host disconnected: '{host}'",
-                    host = ctrlr.hostnqn()
-                );
-
-                if let Some(nex) = nexus {
-                    nex.rm_initiator(&ctrlr.hostnqn());
-                    subsystem.host_disconnect_nexus(ctrlr);
-                } else {
-                    subsystem.host_disconnect_replica(ctrlr);
-                }
-            }
-            NvmfSubsystemEvent::HostKeepAliveTimeout(ctrlr) => {
+        debug!("NVMF subsystem event {s:?}: {event:?}");
+
+        use NvmfSubsystemEvent::*;
+
+        match NqnTarget::lookup(&s.get_nqn()) {
```

Comment:
@datacore-vvarakantham I think Host events might be useful to add to the event bus?

Reply:
Yes. Should I add the category "Host" and actions "Connect", "Disconnect", "KeepAliveTimeout" ?

Reply:
Maybe `HostInitiator`? Otherwise sounds good to me

Reply:
Ok.